deepchecks/tabular/checks/model_evaluation/performance_bias.py

```diff
@@ -0,0 +1,549 @@
+
+from deepchecks.core import CheckResult, ConditionCategory, ConditionResult
+from deepchecks.core.checks import DatasetKind
+from deepchecks.core.errors import DeepchecksProcessError, DeepchecksValueError
+from deepchecks.tabular import Context, SingleDatasetCheck
+from deepchecks.utils.performance.partition import partition_column
+from deepchecks.utils.typing import Hashable
+
+
+class PerformanceBias(SingleDatasetCheck):
+    """
+    Check for performance disparities across subgroups, while optionally accounting for a control variable.
+
+    The check identifies "performance biases": large performance difference for a subgroup compared
+    to the baseline performance for the full population. It is intended to be used for fairness analyses.
+
+    Subgroups are defined by the categories of a "protected" feature. Numerical features are first binned
+    into quantiles, while categorical features are preserved as-is. The baseline score is the overall
+    score when all subgroups are combined. You can add conditions to flag performance differences outside
+    of given bounds. A visual display is also provided to identify subgroups with the largest performance
+    disparities.
+
+    Additionally, the analysis may be separated across the categories of a "control" feature. Numerical
+    features are binned and categorical features are re-binned into `max_number` categories. To account
+    for the control feature, baseline scores and subgroup scores are be computed within each of its
+    categories.
+
+    The check's display is a plot of the baseline score (black point) and subgroup scores (white point)
+    for each subgroup, faceted by control feature categories, and sorted by the size of the
+    performance disparity. The performance disparity is highlighted by a red line for negative differences
+    and a green line for positive differences. By default, only the top 3 categories of the control
+    feature with the largest performance disparities are displayed. Within each control category
+    subplot, the top 3 subgroups with the largest performance disparities are displayed.
+
+    Parameters
+    ----------
+    protected_feature : Hashable
+        Feature evaluated for performance disparities. Numerical features are binned into `max_categories`
+        quantiles. Categorical features are not transformed.
+    control_feature : Hashable, default: None
+        Feature used to group data prior to evaluating performance disparities (disparities are only
+        assessed within the groups defined by this feature). Numerical features are binned into
+        `max_categories` quantiles. Categorical features are re-grouped into at most `max_categories`
+        groups if necessary.
+    scorer : str or Tuple[str, Union[str, Callable]], default: None
+        Name of the performance score function to use.
+    max_categories : int, default: 10
+        Maximum number of categories into which `control_feature` is binned.
```

Comment:
confusing, we usually use max_categories as the maximum number of categories we allow. Should be max_bins